src/textures/CompressedTexture.js

```diff
@@ -1,28 +1,27 @@
 import { Texture } from './Texture.js';
 
-function CompressedTexture( mipmaps, width, height, format, type, mapping, wrapS, wrapT, magFilter, minFilter, anisotropy, encoding ) {
+class CompressedTexture extends Texture {
 
-	Texture.call( this, null, mapping, wrapS, wrapT, magFilter, minFilter, format, type, anisotropy, encoding );
+	constructor( mipmaps, width, height, format, type, mapping, wrapS, wrapT, magFilter, minFilter, anisotropy, encoding ) {
 
-	this.image = { width: width, height: height };
-	this.mipmaps = mipmaps;
+		super( null, mapping, wrapS, wrapT, magFilter, minFilter, format, type, anisotropy, encoding );
+		this.image = { width: width, height: height };
+		this.mipmaps = mipmaps;
 
-	// no flipping for cube textures
-	// (also flipping doesn't work for compressed textures )
+		// no flipping for cube textures
+		// (also flipping doesn't work for compressed textures )
 
-	this.flipY = false;
+		this.flipY = false;
 
-	// can't generate mipmaps for compressed textures
-	// mips must be embedded in DDS files
+		// can't generate mipmaps for compressed textures
+		// mips must be embedded in DDS files
 
-	this.generateMipmaps = false;
+		this.generateMipmaps = false;
 
-}
+	}
 
-CompressedTexture.prototype = Object.create( Texture.prototype );
-CompressedTexture.prototype.constructor = CompressedTexture;
+}
 
 CompressedTexture.prototype.isCompressedTexture = true;
```

Comment:
could you use the `Object.defineProperty()` pattern demonstrated in #20110 across the files modified in this PR? 🙏

Also support for class fields is coming soon.

Reply:
on it